crates/wit-component/src/validation.rs

```diff
@@ -114,6 +117,26 @@ pub enum ImportInstance {
     Names(IndexMap<String, Import>),
 }
 
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub struct PayloadInfo {
+    pub name: String,
+    pub ty: TypeId,
+    pub function: Function,
+    pub key: WorldKey,
+    pub interface: Option<InterfaceId>,
+    pub imported: bool,
+}
```

Comment:
Mind adding some comments to this? I'm this far through wit-component is and I'm still not entirely sure what a "payload" is other than that it's async-related